### PR TITLE
Shutdown taipy gracefully without long traceback

### LINKED ISSUE #1334: [DOCS] How to stop/ shutdown taipy gracefully without a long traceback?
### Issue Description

Right now, every time I close taipy using CTRL+C, a long traceback gets printed.  What is the proper way to close the app without having this? 

### Screenshots or Examples (if applicable)

**^C** is where the interrupt happens
```
poetry run taipy run DataCheckDashboard.py
[2024-05-29 10:58:09][Taipy][INFO] Application is running in 'debug' mode
[2024-05-29 10:58:09][Taipy][INFO] 'allow_unsafe_werkzeug' has been set to True
[2024-05-29 10:58:09][Taipy][INFO] 'async_mode' parameter has been overridden to 'threading'. Using Flask built-in development server with debug mode
[2024-05-29 10:58:09][Taipy][INFO]  * Server starting on http://127.0.0.1:2453
/home/david/miniconda3/envs/onexus_backend_py38/lib/python3.8/subprocess.py:946: ResourceWarning:

subprocess 1302237 is still running

 * Serving Flask app 'Taipy'
 * Debug mode: on
[2024-05-29 10:58:11][Taipy][INFO]  * Server reloaded on http://127.0.0.1:2453
**^C** /home/david/miniconda3/envs/onexus_backend_py38/lib/python3.8/subprocess.py:946: ResourceWarning:

subprocess 1302241 is still running

Traceback (most recent call last):
  File "/home/david/miniconda3/envs/onexus_backend_py38/bin/taipy", line 8, in <module>
    sys.exit(_entrypoint())
  File "/home/david/miniconda3/envs/onexus_backend_py38/lib/python3.8/site-packages/taipy/_entrypoint.py", line 53, in _entrypoint
    _RunCLI.parse_arguments()
  File "/home/david/miniconda3/envs/onexus_backend_py38/lib/python3.8/site-packages/taipy/_cli/_run_cli.py", line 53, in parse_arguments
    subprocess.run(
  File "/home/david/miniconda3/envs/onexus_backend_py38/lib/python3.8/subprocess.py", line 495, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
  File "/home/david/miniconda3/envs/onexus_backend_py38/lib/python3.8/subprocess.py", line 1020, in communicate
    self.wait()
  File "/home/david/miniconda3/envs/onexus_backend_py38/lib/python3.8/subprocess.py", line 1083, in wait
    return self._wait(timeout=timeout)
  File "/home/david/miniconda3/envs/onexus_backend_py38/lib/python3.8/subprocess.py", line 1806, in _wait
    (pid, sts) = self._try_wait(0)
  File "/home/david/miniconda3/envs/onexus_backend_py38/lib/python3.8/subprocess.py", line 1764, in _try_wait
    (pid, sts) = os.waitpid(self.pid, wait_flags)
KeyboardInterrupt
/home/david/miniconda3/envs/onexus_backend_py38/lib/python3.8/subprocess.py:946: ResourceWarning: subprocess 1302157 is still running
  _warn("subprocess %s is still running" % self.pid,
```

### Proposed Solution (optional)

Either document how to shut it down gracefully or just show `shutting down server` without showing traceback.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by PrathamGupta06**
Fixes #1334 

On pressing ctrl + c, keyboard interrupt error is not shown now.
image

### CODE REVIEW

**Comment by trgiangdo** [comment]
As discussed in the issue, it doesn't seem to fix the problem at all. Can you share your main script?

**Comment by PrathamGupta06** [comment]
It does fix the problem. I used this script to test the fix
```py
# main.py
from taipy import Gui

Gui(page="# Getting started with *Taipy*").run(debug=True) # use_reloader=True
```

I can see the KeyboardInterrupt error before this fix and it disappears after this fix.  
My procedure to test this was: 
- First I ran `python -m pip install taipy/` to install from the taipy directory.
- `taipy run main.py`
- I hit ctrl + c and I see the errors.
- I added Exception Handling for KeyboardInterrupt.
- Again ran `python -m pip install taipy/`
- `taipy run main.py`
- I hit ctrl + c and I don't see the errors any more.

**Comment by trgiangdo** [comment]
It's because you are starting the GUI service with `debug=True`, which is False by default.

Correct me if I'm wrong.
When you have debug mode on, the async_mode will be set to `"threading"`, which tells Socket.IO to use the Python threading model for handling asynchronous events. The try-catch is enough in this case.

By default, the async_mode is `gevent`. Then the `KeyboardInterrupt` exception will be handled in the event loop before reaching the outer try-catch.

**Comment by PrathamGupta06** [comment]
Yes, you are correct. Even after using a signal handler, gevent is handling the exception before coming out of the event loop. 
Does this would mean that unless we modify the gevent source code itself, this is not resolvable?

**Comment by trgiangdo** [comment]
Probably we need to put the try catch elsewhere. Needs more debugging and experimenting

**Comment by dinhlongviolin1** [comment]
For gevent you can look into this link. Might be helpful

https://stackoverflow.com/questions/45556949/python-gevent-unexpected-output-in-keyboardinterrupt

**Comment by PrathamGupta06** [comment]
> For gevent you can look into this link. Might be helpful
> 
> https://stackoverflow.com/questions/45556949/python-gevent-unexpected-output-in-keyboardinterrupt

Thank you, will look into it

**Comment by PrathamGupta06** [comment]
KeyboardInterrupt doesn't appear now even without debug. Thanks @dinhlongviolin1

File: taipy/gui/server.py [hunk 1]
```diff
@@ -256,8 +256,8 @@ def _get_async_mode(self) -> str:
 
     def _apply_patch(self):
         if self._get_async_mode() == "gevent" and util.find_spec("gevent"):
-            from gevent import monkey
-
+            from gevent import monkey, get_hub
```
Comment (FabienLelaquais) [inline taipy/gui/server.py]:
```suggestion
            from gevent import get_hub, monkey

```
for the linter (including the empty line)

**Review by FabienLelaquais — DISMISSED** [review]
A tiny formatting change that is mandary.
Nice job!

**Comment by FabienLelaquais** [comment]
Well done, @PrathamGupta06, and thank you, @dinhlongviolin1!
That seems to do the trick.

I'm running the tests - I know just by reading the code that they will fail until you sort the imports - no big deal though.

**Review by FabienLelaquais — DISMISSED** [review]
This looks great!

**Comment by trgiangdo** [comment]
This's looking good.
For the failing test, there's just some redundant blank spaces. Please check the linter test for more detail @PrathamGupta06

**Comment by FabienLelaquais** [comment]
@trgiangdo is completely right.
I just committed the changes to adjust for the tests. Thanks!

**Review by trgiangdo — APPROVED** [review]

**Review by dinhlongviolin1 — APPROVED** [review]
LGTM

**Comment by dinhlongviolin1** [comment]
@PrathamGupta06 thanks for the PR. We will proceed to merge the PR 

@FabienLelaquais please do the honor of reviewing and merging it

**Review by FabienLelaquais — APPROVED** [review]
Perfect!

**Comment by FabienLelaquais** [comment]
@PrathamGupta06 this pull request is now part of the Taipy code base!
Thanks a lot!!

**Comment by jrobinAV** [comment]
@PrathamGupta06 Thx for this!

**Comment by PrathamGupta06** [comment]
My pleasure @jrobinAV 😊 Looking forward to more contributions in taipy.

**Pull request merged**